### DISCUSSION TITLE
Where is the f-string part for the Python grammar

### DISCUSSION THREAD

**Discussion by ThatXliner** [lark-parser/lark#886] [Q&A] [answered]
Hello! I'm trying to ~~steal~~ ~~copy~~ borrow the grammar for Python's [f-strings](https://www.python.org/dev/peps/pep-0498/). Currently, I can't find it on the [grammar file](https://github.com/lark-parser/lark/blob/master/examples/advanced/python3.lark). Is this implemented or did I skip it somewhere?

**Comment by MegaIng** [comment, marked as answer]
It is 'implemented' the same way it is 'implemented' in the official python grammar: A later parser pass separates the corresponding part out of the larger string and then does with it what it needs to do and calls back the original parser. It is not actually implemented in the base grammar file.

That doesn't mean it can't be. The problem is just that then the string couldn't be a single token, which would probably complicated the resulting Tree and slow down parsing.

**Reply by ThatXliner** [reply]
Oh, I see! So it post-parses the string, kind of like a shell's `interpolation="interpolation" echo "string $interpolation"`!
